tests/unit/test_stages/test_base_stage.py

```diff
@@ -187,6 +200,35 @@ def test_dashboard_view_filters_students(self, workgroups, target_students, filt
 
         patched_stats.assert_called_once_with(workgroups, expected_students)
 
+    @ddt.data(
+        ([1], [1], [], make_stats(1, 0, 0), True),
+        (range(10), range(3), range(3, 6), make_stats(.3, .3, .4), True),
+        ([], [], [], make_stats(None, None, None), False),
+    )
+    @ddt.unpack
+    def test_get_stage_stats(
+            self, all_user_ids, completed_user_ids, partial_user_ids, expected_stats, completions_called):
+        all_users = [make_reduced_user_details(id=id) for id in all_user_ids]
+        completed_user_ids = set(completed_user_ids)
+        partial_user_ids = set(partial_user_ids)
+        patched_completions = self.make_patch(self.block, 'get_users_completion')
+        self.block.display_name = "dummy block"
+        patched_completions.return_value = (completed_user_ids, partial_user_ids)
+
+        # marker object
+        target_workgroups = object()
```

Comment:
@jbzdak I believe it would be better if it was something more like actual value, i.e. mock.Mock(spec=list), or sometihn glike that.